x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -660,6 +671,52 @@ public void testMountedSnapshotHasNoReplicasByDefault() throws Exception {
         }
     }
 
+    public void testSnapshotMountedIndexLeavesBlobsUntouched() throws Exception {
+        final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        createAndPopulateIndex(
+            indexName,
+            Settings.builder()
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, between(1, 3))
+                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                .put(INDEX_SOFT_DELETES_SETTING.getKey(), true)
+        );
+        ensureGreen(indexName);
+        forceMerge();
+
+        final String repositoryName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
+        final Path repositoryLocation = randomRepoPath();
+        createFsRepository(repositoryName, repositoryLocation);
+
+        final SnapshotId snapshotOne = createSnapshot(repositoryName, List.of(indexName));
+        final Set<Tuple<String, Long>> snapshotOneDataBlobs = listDataBlobsInRepository(repositoryLocation);
+        assertThat(snapshotOneDataBlobs, hasSize(greaterThan(0)));
+        assertAcked(client().admin().indices().prepareDelete(indexName));
+
+        mountSnapshot(repositoryName, snapshotOne.getName(), indexName, indexName, Settings.EMPTY);
+        ensureGreen(indexName);
+
+        createSnapshot(repositoryName, List.of(indexName));
+        final Set<Tuple<String, Long>> snapshotTwoDataBlobs = listDataBlobsInRepository(repositoryLocation);
+        assertThat(snapshotTwoDataBlobs.containsAll(snapshotOneDataBlobs), is(true));
+        assertThat(snapshotTwoDataBlobs, hasSize(snapshotOneDataBlobs.size()));
```

Comment:
I think you can have this check in a more straight-forward manner and just use the snapshot status API.

See for example `org.elasticsearch.snapshots.BlobStoreIncrementalityIT#assertTwoIdenticalShardSnapshots`. I think I'd prefer that over adding yet another test that manually inspects the exact file structure on disk and has to be adjusted if and when we change the repo format.
In a sense this is also more correct since we already store some blobs straight in the shard snapshot metadata where the approach would catch changes while the data files don't change.

Reply:
I hadn't even thought about it, thanks for the suggestion